src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -234,13 +235,94 @@ public AndroidTzData()
                     {
                         _tzFileDir = tzFileDir;
                         _tzFilePath = tzFilePath;
+                        _isFiltered = false;
                         return;
                     }
                 }
 
                 throw new TimeZoneNotFoundException(SR.TimeZoneNotFound_ValidTimeZoneFileMissing);
             }
 
+            // On some versions of Android, the tzdata file may still contain backward timezone ids.
+            // We attempt to use tzlookup.xml, which is available on some versions of Android to help
+            // validate non-backward timezone ids
+            // tzlookup.xml is an autogenerated file that contains timezone ids in this form:
+            //
+            // <timezones ianaversion="2019b">
+            //   <countryzones>
+            //     <country code="au" default="Australia/Sydney" everutc="n">
+            //       <id alts="Australia/ACT,Australia/Canberra,Australia/NSW">Australia/Sydney</id>
+            //       ...
+            //       ...
+            //       <id>Australia/Eucla</id>
+            //     </country>
+            //     <country ...>
+            //       ...
+            //       ...
+            //       ...
+            //     </country>
+            //   </countryzones>
+            // </timezones>
+            //
+            // Once the timezone cache is populated with the IDs, we reference tzlookup id tags and
+            // remove IDs and their associated information (byteoffsets and lengths) from the current
+            // AndroidTzData instance.
+            private void FilterBackwardIDs(string lookupPath)
+            {
+                try
+                {
+                    using (StreamReader sr = File.OpenText(lookupPath))
+                    {
+                        var tzLookupIDs = new HashSet<string>();
+                        string? tzLookupLine;
+                        while (sr.Peek() >= 0)
+                        {
+                            if (!(tzLookupLine = sr.ReadLine())!.TrimStart().StartsWith("<id"))
+                                continue;
+
```

Comment:
Is there not a standard XML parser that's used for XML files?  (Honest question; I have little familiarity with this codebase.)

Reply:
No, not where the code is being changed.  The only option I know of other than this would be to interop with some native library. Would be overkill for this change in my opinion.

Reply:
Got it, thx.